### PR TITLE
Dvoss/microsoft edge cases

### PR DESCRIPTION
**Pull request by dvosscricut**
Fix plus added regression test for Protocols exported to Dart that have a leading underscore in their property names

### CODE REVIEW

File: Sources/FishyJoesCore/Unparse/DartClass.swift [hunk 1]
```diff
@@ -235,7 +234,7 @@ class DartClass {
                 func outputParameter(parameter: Method.Parameter) {
                     let labelComment = parameter.labelComment.map { "/* \($0) */ " } ?? ""
                     let defaultValue = parameter.defaultValue.map { " = \($0)" } ?? ""
-                    fragment.output("\(parameter.type.name(in: self)) \(labelComment)\(DartClass.deforbidify(parameter.name))\(defaultValue),")
+                    fragment.output("\(parameter.type.name(in: self)) \(labelComment)\(Self.deforbidify(parameter.name))\(defaultValue),")
```
Comment (cobbal) [inline Sources/FishyJoesCore/Unparse/DartClass.swift]: "Self" should be avoided in classes unless you actually want the dynamic type of the current instance. Since there seems to be no overloads of "deforbidify", it should remain "DartClass"

**Review by cobbal — CHANGES_REQUESTED** [review]

**Review by mstokercricut — APPROVED** [review]
Great new tests.

File: Sources/FishyJoesCore/Unparse/DartClass.swift (hunk 1 above)
Comment (dvosscricut) [reply Sources/FishyJoesCore/Unparse/DartClass.swift]: ok

**Review by cobbal — APPROVED** [review]
Looks good to me if CI passes (nothing windows-specific in here, so that one check can be ignored due to flakiness for now)
